packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -111,25 +123,74 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
     event WithdrawalFinalized(bytes32 indexed withdrawalHash, bool success);
 
     /**
-     * @custom:semver 1.0.0
+     * @notice Emitted when the pause is triggered.
+     *
+     * @param account Address of the account triggering the pause.
+     */
+    event Paused(address account);
+
+    /**
+     * @notice Emitted when the pause is lifted.
+     *
+     * @param account Address of the account triggering the unpause.
+     */
+    event Unpaused(address account);
+
+    /**
+     * @notice Reverts when paused.
+     */
+    modifier whenNotPaused() {
+        require(paused == false, "OptimismPortal: paused");
+        _;
+    }
+
+    /**
+     * @custom:semver 1.1.0
      *
      * @param _l2Oracle                  Address of the L2OutputOracle contract.
+     * @param _guardian                  Address that can pause deposits and withdrawals.
      * @param _finalizationPeriodSeconds Output finalization time in seconds.
+     * @param _paused                    Sets the contract's pausability state.
      */
-    constructor(L2OutputOracle _l2Oracle, uint256 _finalizationPeriodSeconds) Semver(1, 0, 0) {
+    constructor(
+        L2OutputOracle _l2Oracle,
+        address _guardian,
+        uint256 _finalizationPeriodSeconds,
+        bool _paused
```

Comment:
Very weak preference to keep these related items together:
```suggestion
        L2OutputOracle _l2Oracle,
        uint256 _finalizationPeriodSeconds,
        address _guardian,
        bool _paused
```

Or maybe create a type? 

```
struct PausableConfig {
    address _guardian;
    bool _paused;
}
```

Reply:
Preference for not a struct so the abi is more clear but get your point on grouping, agreed on that